app/js/lib/ng_utils.js

```diff
@@ -1241,8 +1241,8 @@ angular.module('izhukov.utils', [])
     var soundcloudRegExp = /^https?:\/\/(?:soundcloud\.com|snd\.sc)\/([a-zA-Z0-9%\-\_]+)\/([a-zA-Z0-9%\-\_]+)/i
     var spotifyRegExp = /(https?:\/\/(open\.spotify\.com|play\.spotify\.com|spoti\.fi)\/(.+)|spotify:(.+))/i
 
-    var markdownTestRegExp = /[`_*@]/
-    var markdownRegExp = /(^|\s|\n)(````?)([\s\S]+?)(````?)([\s\n\.,:?!;]|$)|(^|\s)(`|\*\*|__)([^\n]+?)\7([\s\.,:?!;]|$)|@(\d+)\s*\((.+?)\)/m
+    var markdownTestRegExp = /[`_*@]|\]\(/
+    var markdownRegExp = new RegExp('(^|\\s|\\n)(````?)([\\s\\S]+?)(````?)([\\s\\n\\.,:?!;]|$)|(^|\\s)(`|\\*\\*|__)([^\\n]+?)\\7([\\s\\.,:?!;]|$)|@(\\d+)\\s*\\((.+?)\\)|\\[([^\\]]+)\\]\\((' + urlRegExp + ')\\)', 'm')
```

Comment:
Since it may be a bit hard to see what's going on here, let's go through it:
basically this just adds the regex part for matching text links at the very end - pseudo regex: `/\[([^\]]+)\]\((urlRegExp)\)/`.

To be able to insert and reuse `urlRegExp` from above the literal regex had to be converted to a string - to be concatenated with `urlRegExp` - that is then passed to the `RegExp` constructor, i.e. the delimiting `/`s were removed/replaced by single quotes and all back slashes had to be escaped (`\` -> `\\`).

This adds two more (relevant) capturing groups:
* `match[12]`: the link text
* `match[13]`: the url